src/Application/GamePathResolver.cpp

```diff
@@ -3,94 +3,115 @@
         "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/GOG.com/Games/1207661253/PATH",
         "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/GOG.com/GOGMM6/PATH",
-        "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/New World Computing/Might and Magic\xC2\xAE VI/1.0/AppPath",
+        "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/New World Computing/Might and Magic\xC2\xAE VI/1.0/AppPath"
     }
 };
 
-static constexpr PathResolutionConfig mm7Config = {
+
+constexpr PathResolutionConfig mm7Config {
     mm7PathOverrideKey,
     {
         "HKEY_LOCAL_MACHINE/SOFTWARE/GOG.com/Games/1207658916/Path",
         "HKEY_LOCAL_MACHINE/SOFTWARE/GOG.com/GOGMM7/PATH",
         "HKEY_LOCAL_MACHINE/SOFTWARE/New World Computing/Might and Magic VII/1.0/AppPath",
         "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/GOG.com/Games/1207658916/Path",
         "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/GOG.com/GOGMM7/PATH",
-        "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/New World Computing/Might and Magic VII/1.0/AppPath",
+        "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/New World Computing/Might and Magic VII/1.0/AppPath"
     }
 };
 
-static constexpr PathResolutionConfig mm8Config = {
+
+constexpr PathResolutionConfig mm8Config {
     mm8PathOverrideKey,
     {
         "HKEY_LOCAL_MACHINE/SOFTWARE/GOG.com/GOGMM8/PATH",
         "HKEY_LOCAL_MACHINE/SOFTWARE/New World Computing/Might and Magic Day of the Destroyer/1.0/AppPath",
         "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/GOG.com/GOGMM8/PATH",
-        "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/New World Computing/Might and Magic Day of the Destroyer/1.0/AppPath",
+        "HKEY_LOCAL_MACHINE/SOFTWARE/WOW6432Node/New World Computing/Might and Magic Day of the Destroyer/1.0/AppPath"
     }
 };
 
-static std::vector<std::string> resolvePaths(Environment *environment, const PathResolutionConfig &config) {
+
+std::vector<std::string> resolvePaths(const Environment &environment, const PathResolutionConfig &config) {
     // If we have a path override then it'll be the only path we'll check.
-    std::string envPath = environment->getenv(config.overrideEnvKey);
-    if (!envPath.empty()) {
-        logger->info("Path override provided, '{}={}'.", config.overrideEnvKey, envPath);
-        return {envPath};
+    std::string_view envKey = config.overrideEnvKey;
+    if (std::string envPath = environment.getenv(envKey); !envPath.empty()) {
+        logger->info("Path override provided, '{}={}'.", envKey, envPath);
+        return { envPath };
     }
 
     std::vector<std::string> result;
+#ifdef _WIN32
+    std::size_t pathSize = config.registryKeys.size() + 1;
+#elif __APPLE__
+    constexpr std::size_t pathSize = 2;
+#elif __ANDROID__
+    constexpr std::size_t pathSize = 3;
+#else
+    constexpr std::size_t pathSize = 1;
+#endif
 
-    // Otherwise we check PWD first.
-    result.push_back(".");
+    result.reserve(pathSize);
```

Comment:
This is not really needed

Reply:
Why `std::vector::reserve() ` is unnecessary?

Reply:
We are not growing the vector enough to get any meaningful use of reserve optimization - all this does is add extra platform specific code to maintain.